🧑‍🌾 rosbag2_transport windows warnings

Reference builds:
* https://ci.ros2.org/job/nightly_win_rep/3652/
* https://ci.ros2.org/job/nightly_win_rel/3346/

Msbuild warnings:

```
  player_progress_bar.cpp

  Defining YAML_CPP_API for DLL import

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\include\rosbag2_transport/player_progress_bar.hpp(91,42): warning C4251: 'rosbag2_transport::PlayerProgressBar::pimpl_': class 'std::unique_ptr<rosbag2_transport::PlayerProgressBarImpl,std::default_delete<rosbag2_transport::PlayerProgressBarImpl>>' needs to have dll-interface to be used by clients of class 'rosbag2_transport::PlayerProgressBar' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\src\rosbag2_transport\player_progress_bar.cpp) [C:\ci\ws\build\rosbag2_transport\rosbag2_transport.vcxproj]

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\include\rosbag2_transport/player_progress_bar.hpp(91): message : see declaration of 'std::unique_ptr<rosbag2_transport::PlayerProgressBarImpl,std::default_delete<rosbag2_transport::PlayerProgressBarImpl>>' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\src\rosbag2_transport\player_progress_bar.cpp) [C:\ci\ws\build\rosbag2_transport\rosbag2_transport.vcxproj]

  Defining YAML_CPP_API for DLL import

  player_service_client.cpp

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\include\rosbag2_transport/player_progress_bar.hpp(91,42): warning C4251: 'rosbag2_transport::PlayerProgressBar::pimpl_': class 'std::unique_ptr<rosbag2_transport::PlayerProgressBarImpl,std::default_delete<rosbag2_transport::PlayerProgressBarImpl>>' needs to have dll-interface to be used by clients of class 'rosbag2_transport::PlayerProgressBar' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\src\rosbag2_transport\player.cpp) [C:\ci\ws\build\rosbag2_transport\rosbag2_transport.vcxproj]

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\include\rosbag2_transport/player_progress_bar.hpp(91): message : see declaration of 'std::unique_ptr<rosbag2_transport::PlayerProgressBarImpl,std::default_delete<rosbag2_transport::PlayerProgressBarImpl>>' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\src\rosbag2_transport\player.cpp) [C:\ci\ws\build\rosbag2_transport\rosbag2_transport.vcxproj]

...

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\include\rosbag2_transport/player_progress_bar.hpp(91): message : see declaration of 'std::unique_ptr<rosbag2_transport::PlayerProgressBarImpl,std::default_delete<rosbag2_transport::PlayerProgressBarImpl>>' [C:\ci\ws\build\rosbag2_transport\test_player_progress_bar.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include\memory(3381,59): warning C4267: 'argument': conversion from 'size_t' to 'int32_t', possible loss of data [C:\ci\ws\build\rosbag2_transport\test_player_progress_bar.vcxproj]
```

@Crola1702 Thanks for reporting these warnings.  I haven't seen them on a regular CI run.
I will address the following warnings in the `rosbag2_progress_bar`
```
C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\include\rosbag2_transport/player_progress_bar.hpp(91,42): warning C4251: 'rosbag2_transport::PlayerProgressBar::pimpl_': class 'std::unique_ptr<rosbag2_transport::PlayerProgressBarImpl,std::default_delete<rosbag2_transport::PlayerProgressBarImpl>>' needs to have dll-interface to be used by clients of class 'rosbag2_transport::PlayerProgressBar' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\src\rosbag2_transport\player_progress_bar.cpp) [C:\ci\ws\build\rosbag2_transport\rosbag2_transport.vcxproj]

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\include\rosbag2_transport/player_progress_bar.hpp(91): message : see declaration of 'std::unique_ptr<rosbag2_transport::PlayerProgressBarImpl,std::default_delete<rosbag2_transport::PlayerProgressBarImpl>>' (compiling source file C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\src\rosbag2_transport\player_progress_bar.cpp) [C:\ci\ws\build\rosbag2_transport\rosbag2_transport.vcxproj]
```
and 
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include\memory(3381,59): warning C4267: 'argument': conversion from 'size_t' to 'int32_t', possible loss of data [C:\ci\ws\build\rosbag2_transport\test_player_progress_bar.vcxproj]
```
**However, I don't see other warnings**
```
  player_progress_bar.cpp

  Defining YAML_CPP_API for DLL import
```
and 
```
  Defining YAML_CPP_API for DLL import

  player_service_client.cpp
```
in the linked CI jobs
- https://ci.ros2.org/job/nightly_win_rep/3652/
- https://ci.ros2.org/job/nightly_win_rel/3346/

--- a/rosbag2_transport/include/rosbag2_transport/player_progress_bar.hpp
+++ b/rosbag2_transport/include/rosbag2_transport/player_progress_bar.hpp
@@ -21,6 +21,20 @@
 #include "rcutils/time.h"
 #include "rosbag2_transport/visibility_control.hpp"
 
+#ifdef _WIN32
+#  pragma warning(push)
+// Suppress warning "rosbag2_transport::PlayerProgressBar::pimpl_': class 'std::unique_ptr>' needs
+// to have dll-interface to be used by clients of class 'rosbag2_transport::PlayerProgressBar'"
+// Justification:
+// 1. We never inline code in the header that actually calls methods on PlayerProgressBarImpl.
+// 2. While the `PlayerProgressBarImpl` is defined in the `player_progress_bar_impl.hpp` file, we
+// include it only in the `player_progress_bar.cpp` file, and it does not leak into the external
+// API.
+// 3. The pimpl design pattern imply that implementation details are hidden and shouldn't be
+// exposed with the dll-interface.
+#  pragma warning(disable:4251)
+#endif
+
 namespace rosbag2_transport
 {
 class PlayerProgressBarImpl;
@@ -92,5 +106,9 @@ private:
 };
 
 }  // namespace rosbag2_transport
+
+#ifdef _WIN32
+#  pragma warning(pop)
+#endif
 
 #endif  // ROSBAG2_TRANSPORT__PLAYER_PROGRESS_BAR_HPP_

--- a/rosbag2_transport/test/rosbag2_transport/test_player_progress_bar.cpp
+++ b/rosbag2_transport/test/rosbag2_transport/test_player_progress_bar.cpp
@@ -226,10 +226,10 @@ TEST_F(TestPlayerProgressBar, update_with_limited_rate_respect_update_rate) {
 
   // Test if update rate of the progress bar is respected
   std::ostringstream oss;
-  const size_t update_rate = 4;
+  const int32_t update_rate = 4;
   const rcutils_duration_value_t update_period_ns = RCUTILS_S_TO_NS(1) / update_rate;
   auto progress_bar =
-    std::make_unique<PlayerProgressBar>(oss, starting_time, ending_time, update_rate, 0);
+    std::make_unique<PlayerProgressBar>(oss, starting_time, ending_time, update_rate, 0U);
 
   std::vector<rcutils_time_point_value_t> timestamps;
   timestamps.push_back(starting_time);  // 1st update of the progress bar

